airflow/executors/base_executor.py

```diff
@@ -55,7 +55,7 @@ class BaseExecutor(LoggingMixin):
         ``0`` for infinity
     """
 
-    job_id: Optional[str] = None
+    job_id: Optional[int] = None
```

Comment:
We pass this as a `str` (from Backfill and Manual triggers) and `int` from Scheduler

https://github.com/apache/airflow/blob/5980d2b05eee484256c634d5efae9410265c65e9/airflow/jobs/backfill_job.py#L807

https://github.com/apache/airflow/blob/5980d2b05eee484256c634d5efae9410265c65e9/airflow/jobs/scheduler_job.py#L662

https://github.com/apache/airflow/blob/5980d2b05eee484256c634d5efae9410265c65e9/airflow/jobs/base_job.py#L56

Reply:
I did wonder if this might cause a problem. I guess it should be `int | str | None` then.

Reply:
Yeah

Reply:
Changing this to `int | str | None` affects quite a few things to make the changes non-trivial, so I created a separate PR for easier review: astronomer/airflow#1481.